plugins/database/cassandra/connection_producer.go

```diff
@@ -269,6 +230,50 @@ func (c *cassandraConnectionProducer) createSession(ctx context.Context) (*gocql
 	return session, nil
 }
 
+func getSslOpts(certBundle *certutil.CertBundle, minTLSVersion string, insecureSkipVerify bool) (*gocql.SslOptions, error) {
+	var tlsConfig *tls.Config
+	if certBundle != nil {
+		if certBundle.Certificate == "" && certBundle.PrivateKey != "" {
+			return nil, fmt.Errorf("found private key for TLS authentication but no certificate")
+		}
+		if certBundle.Certificate != "" && certBundle.PrivateKey == "" {
+			return nil, fmt.Errorf("found certificate for TLS authentication but no private key")
+		}
+
+		parsedCertBundle, err := certBundle.ToParsedCertBundle()
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse certificate bundle: %w", err)
+		}
+
+		tlsConfig, err := parsedCertBundle.GetTLSConfig(certutil.TLSClient)
+		if err != nil || tlsConfig == nil {
```

Comment:
Looking at `GetTLSConfig()`, it doesn't seem that we can get a nil `tlsConfig` if the error is nil so a check on `err != nil` should be sufficient.

Reply:
Good catch, fixed.